src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/factoryOrchestrator.hpp

```diff
@@ -50,44 +56,57 @@ class TFactoryOrchestrator final
      */
     static std::shared_ptr<AbstractHandler<std::shared_ptr<TScanContext>>>
     create(ScannerType type,
            std::shared_ptr<TDatabaseFeedManager> databaseFeedManager,
            std::shared_ptr<TIndexerConnector> indexerConnector,
            Utils::RocksDBWrapper& inventoryDatabase,
-           std::shared_ptr<SocketClient<Socket<OSPrimitives, NoHeaderProtocol>, EpollWrapper>> reportSocketClient)
+           std::shared_ptr<SocketClient<Socket<OSPrimitives, NoHeaderProtocol>, EpollWrapper>> reportSocketClient,
+           std::shared_ptr<AbstractHandler<std::shared_ptr<TScanContext>>> subOrchestration = nullptr)
     {
         std::shared_ptr<AbstractHandler<std::shared_ptr<TScanContext>>> orchestration;
-        if (type == ScannerType::PackageInsert)
+        switch (type)
         {
-            orchestration = std::make_shared<TPackageScanner>(databaseFeedManager);
-            orchestration->setLast(std::make_shared<TDetailsAugmentation>(databaseFeedManager));
-        }
-        else if (type == ScannerType::PackageDelete)
-        {
-        }
-        else if (type == ScannerType::HotfixInsert)
-        {
-            // orchestration = std::make_shared<HotfixScanner>();
-        }
-        else if (type == ScannerType::HotfixDelete)
-        {
-            // orchestration = std::make_shared<DeleteHotfixVulnerabilities>();
-        }
-        else if (type == ScannerType::Os)
-        {
-            orchestration = std::make_shared<TOsScanner>();
-            // orchestration->setLast(std::make_shared<CheckRemediations>());
-        }
-        else if (type == ScannerType::IntegrityClear)
-        {
-            // orchestration = std::make_shared<VulnerabilityCleanupController>();
-        }
-        else
-        {
-            throw std::runtime_error("Invalid scanner type");
+            case ScannerType::PackageInsert:
+                orchestration = std::make_shared<TPackageScanner>(databaseFeedManager);
+                orchestration->setLast(std::make_shared<TDetailsAugmentation>(databaseFeedManager));
+                break;
+
+            case ScannerType::PackageDelete:
+                // Add implementation for PackageDelete
+                break;
+
+            case ScannerType::HotfixInsert:
+                // orchestration = std::make_shared<HotfixScanner>();
+                break;
+
+            case ScannerType::HotfixDelete:
+                // orchestration = std::make_shared<DeleteHotfixVulnerabilities>();
+                break;
+
+            case ScannerType::Os:
+                orchestration = std::make_shared<TOsScanner>();
+                // orchestration->setLast(std::make_shared<CheckRemediations>());
+                break;
+
+            case ScannerType::IntegrityClear:
+                // orchestration = std::make_shared<VulnerabilityCleanupController>();
+                break;
+
+            case ScannerType::CleanupPackages:
```

Comment:
We should allow a full clean that simply removes everything from the inventory without getting IDs from Wazuh-DB

Reply:
We can left this for another issue

- #21661

Reply:
I agree that this feature requires more issue.
If we define the scope of each of them clearly, then the review will be easier.

Reply:
Fix it!